opentracing-impl-java8/src/main/java/io/opentracing/AbstractTracer.java

```diff
@@ -36,20 +34,20 @@ public SpanBuilder buildSpan(String operationName){
     }
 
     @Override
-    public void inject(SpanContext spanContext, Object carrier) {
-        registry.getInjector(carrier.getClass()).inject(spanContext, carrier);
+    public <C> void inject(SpanContext spanContext, Format<C> format, C carrier) {
```

Comment:
all of this generics stuff can come back now... it was lost in #32. It *does* serve a useful purpose; the compiler/IDE is able to enforce constraints about particular carrier formats.